# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

In [2]:
https://lucid.app/lucidchart/36c47fee-eefb-4009-a469-d4eb57501f13/edit?invitationId=inv_505b5420-255e-40b4-a258-ddd3d86acec8


: Msg 102, Level 15, State 1, Line 1
Incorrect syntax near '/'.

Total execution time: 00:00:00.346

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees?</span>

In [9]:
SELECT * 
FROM LaborStatisticsDB.dbo.datatype
WHERE data_type_text = 'Women employees'


(1 row affected)

Total execution time: 00:00:00.074

data_type_code data_type_text 10 WOMEN EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?</span>

In [18]:
SELECT * 
FROM LaborStatisticsDB.dbo.series
WHERE series_title = 'Women employees'
AND industry_code = '55522110'

SELECT industry_code, industry_name 
FROM LaborStatisticsDB.dbo.industry
WHERE industry_name LIKE '%commercial_bank%'

SELECT * 
FROM LaborStatisticsDB.dbo.supersector
WHERE supersector_name LIKE 'Financial activities'

(2 rows affected)

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.116

series_id supersector_code industry_code data_type_code seasonal series_title CES5552211010 55 55522110 10 S Women employees CEU5552211010 55 55522110 10 U Women employees

industry_code industry_name 55522110 Commercial banking

supersector_code supersector_name 55 Financial activities

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [9]:
SELECT TOP 100 *
FROM LaborStatisticsDB.dbo.annual_2016 as a                                                                                                                                                                                   
LEFT JOIN LaborStatisticsDB.dbo.series as s 
ON a.series_id = s.series_id

(100 rows affected)

Total execution time: 00:00:00.318

id series_id year period value footnote_codes original_file series_id supersector_code industry_code data_type_code seasonal series_title 8755 CEU0000000001 2016 M13 144306 NULL ce.data.0.AllCESSeries.csv CEU0000000001 0 00000000 1 U All employees 22220 CEU0000000001 2016 M13 144306 NULL ce.data.00a.TotalNonfarm.Employment.csv CEU0000000001 0 00000000 1 U All employees 22221 CEU0000000010 2016 M13 71492 NULL ce.data.00a.TotalNonfarm.Employment.csv CEU0000000010 0 00000000 10 U Women employees 8756 CEU0000000010 2016 M13 71492 NULL ce.data.0.AllCESSeries.csv CEU0000000010 0 00000000 10 U Women employees 8757 CEU0500000001 2016 M13 122083 NULL ce.data.0.AllCESSeries.csv CEU0500000001 5 05000000 1 U All employees 6281 CEU0500000001 2016 M13 122083 NULL ce.data.05a.TotalPrivate.Employment.csv CEU0500000001 5 05000000 1 U All employees 7954 CEU0500000002 2016 M13 34.4 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000002 5 05000000 2 U Average weekly hours of all employees 8758 CEU0500000002 2016 M13 34.4 NULL ce.data.0.AllCESSeries.csv CEU0500000002 5 05000000 2 U Average weekly hours of all employees 8759 CEU0500000003 2016 M13 25.65 NULL ce.data.0.AllCESSeries.csv CEU0500000003 5 05000000 3 U Average hourly earnings of all employees 7955 CEU0500000003 2016 M13 25.65 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000003 5 05000000 3 U Average hourly earnings of all employees 6282 CEU0500000006 2016 M13 100531 NULL ce.data.05a.TotalPrivate.Employment.csv CEU0500000006 5 05000000 6 U Production and nonsupervisory employees 8760 CEU0500000006 2016 M13 100531 NULL ce.data.0.AllCESSeries.csv CEU0500000006 5 05000000 6 U Production and nonsupervisory employees 8761 CEU0500000007 2016 M13 33.6 NULL ce.data.0.AllCESSeries.csv CEU0500000007 5 05000000 7 U Average weekly hours of production and nonsupervisory employees 5721 CEU0500000007 2016 M13 33.6 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv CEU0500000007 5 05000000 7 U Average weekly hours of production and nonsupervisory employees 5722 CEU0500000008 2016 M13 21.54 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv CEU0500000008 5 05000000 8 U Average hourly earnings of production and nonsupervisory employees 8762 CEU0500000008 2016 M13 21.54 NULL ce.data.0.AllCESSeries.csv CEU0500000008 5 05000000 8 U Average hourly earnings of production and nonsupervisory employees 8763 CEU0500000010 2016 M13 58747 NULL ce.data.0.AllCESSeries.csv CEU0500000010 5 05000000 10 U Women employees 6283 CEU0500000010 2016 M13 58747 NULL ce.data.05a.TotalPrivate.Employment.csv CEU0500000010 5 05000000 10 U Women employees 7956 CEU0500000011 2016 M13 881.55 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000011 5 05000000 11 U Average weekly earnings of all employees 8764 CEU0500000011 2016 M13 881.55 NULL ce.data.0.AllCESSeries.csv CEU0500000011 5 05000000 11 U Average weekly earnings of all employees 8765 CEU0500000012 2016 M13 367.3 NULL ce.data.0.AllCESSeries.csv CEU0500000012 5 05000000 12 U Average weekly earnings of all employees 7957 CEU0500000012 2016 M13 367.3 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000012 5 05000000 12 U Average weekly earnings of all employees 3808 CEU0500000012 2016 M13 367.3 NULL ce.data.02b.AllRealEarningsAE.csv CEU0500000012 5 05000000 12 U Average weekly earnings of all employees 3809 CEU0500000013 2016 M13 10.69 NULL ce.data.02b.AllRealEarningsAE.csv CEU0500000013 5 05000000 13 U Average hourly earnings of all employees 7958 CEU0500000013 2016 M13 10.69 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000013 5 05000000 13 U Average hourly earnings of all employees 8766 CEU0500000013 2016 M13 10.69 NULL ce.data.0.AllCESSeries.csv CEU0500000013 5 05000000 13 U Average hourly earnings of all employees 8767 CEU0500000016 2016 M13 105.3 NULL ce.data.0.AllCESSeries.csv CEU0500000016 5 05000000 16 U Indexes of aggregate weekly

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [10]:
SELECT TOP 100 *
FROM LaborStatisticsDB.dbo.series as s                                                                                                                                                                                   
JOIN LaborStatisticsDB.dbo.datatype as dt 
ON s.data_type_code = dt.data_type_code

(100 rows affected)

Total execution time: 00:00:00.386

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000010 5 05000000 10 S Women employees 10 WOMEN EMPLOYEES CES0500000011 5 05000000 11 S Average weekly earnings of all employees 11 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000012 5 05000000 12 S Average weekly earnings of all employees 12 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000013 5 05000000 13 S Average hourly earnings of all employees 13 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 16 INDEXES OF AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 17 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES CES0500000019 5 05000000 19 S Average weekly hours of all employees 19 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000021 5 05000000 21 S Diffusion indexes 21 DIFFUSION INDEXES CES0500000022 5 05000000 22 S Diffusion indexes 22 DIFFUSION INDEXES CES0500000023 5 05000000 23 S Diffusion indexes 23 DIFFUSION INDEXES CES0500000025 5 05000000 25 S All employees 25 ALL EMPLOYEES CES0500000026 5 05000000 26 S All employees 26 ALL EMPLOYEES CES0500000030 5 05000000 30 S Average weekly earnings of production and nonsupervisory employees 30 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000031 5 05000000 31 S Average weekly earnings of production and nonsupervisory employees 31 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000032 5 05000000 32 S Average hourly earnings of production and nonsupervisory employees 32 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000034 5 05000000 34 S Indexes of aggregate weekly hours of production and nonsupervisory employees 34 INDEXES OF AGGREGATE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000035 5 05000000 35 S Indexes of aggregate weekly payrolls of production and nonsupervisory employees 35 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000036 5 05000000 36 S Average weekly hours 36 AVERAGE WEEKLY HOURS CES0500000056 5 05000000 56 S Aggregate weekly hours of all employees 56 AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES CES0500000057 5 05000000 57 S Aggregate weekly payrolls of all employees 57 AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES CES0500000081 5 05000000 81 S Aggregate weekly hours of production and nonsupervisory employees 81 AGGREGATE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000082 5 05000000 82 S Aggregate weekly payrolls of production and nonsupervisory employees 82 AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0600000001 6 06000000 1 S All employees 1 ALL EMPLOYEES CES0600000002 6 06000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0600000003 6 06000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES06

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [11]:
SELECT TOP 100 *
FROM LaborStatisticsDB.dbo.series as s                                                                                                                                                                                   
JOIN LaborStatisticsDB.dbo.industry as i 
ON s.industry_code = i.industry_code

(100 rows affected)

Total execution time: 00:00:00.299

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000010 5 05000000 10 S Women employees 1 5000000 - A Total private 1 T 2 CES0500000011 5 05000000 11 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000012 5 05000000 12 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000013 5 05000000 13 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 1 5000000 - A Total private 1 T 2 CES0500000019 5 05000000 19 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000021 5 05000000 21 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000022 5 05000000 22 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000023 5 05000000 23 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000025 5 05000000 25 S All employees 1 5000000 - A Total private 1 T 2 CES0500000026 5 05000000 26 S All employees 1 5000000 - A Total private 1 T 2 CES0500000030 5 05000000 30 S Average weekly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000031 5 05000000 31 S Average weekly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000032 5 05000000 32 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000034 5 05000000 34 S Indexes of aggregate weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000035 5 05000000 35 S Indexes of aggregate weekly payrolls of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000036 5 05000000 36 S Average weekly hours 1 5000000 - A Total private 1 T 2 CES0500000056 5 05000000 56 S Aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000057 5 05000000 57 S Aggregate weekly payrolls of all employees 1 5000000 - A Total private 1 T 2 CES0500000081 5 05000000 81 S Aggregate weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000082 5 05000000 82 S Aggregate weekly payrolls of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0600000001 6 06000000 1 S All employees 2 6000000 - A Goods-producing 1 T 3 CES0600000002 6 06000000 2 S Average weekly hours of all employees 2 6000000 - A Goods-producing 1 T 3 CES0600000003 6 06000000 3 S Average hourly earnings of all employees 2 6000000 - A Goods-producing 1 T 3 CES0600000006 6 06000000 6 S Production and nonsupervisory employees 2 6000000 - A Goods-producing 1 T 3 CES0600000007 6 06000000 7 S Average weekly hours of production and nonsupervisory employees 2 6000000 - A Goods-producing 1 T 3 C

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.</span>

In [14]:
-- SELECT SUM(id)
-- FROM LaborStatisticsDB.dbo.industry
-- WHERE data_type_text = 'ALL EMPLOYEES'


SELECT ROUND(SUM(a.VALUE), 0) AS 'EMPLOYEES ACROSS INDUSTRIES'
FROM LaborStatisticsDB.dbo.annual_2016 AS a 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = a.series_id
WHERE data_type_code = '01'

(1 row affected)

Total execution time: 00:00:00.109

EMPLOYEES ACROSS INDUSTRIES 2340612

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;</span>

In [16]:
SELECT ROUND(SUM(a.VALUE), 0) AS 'WOMEN ACROSS INDUSTRIES'
FROM LaborStatisticsDB.dbo.annual_2016 AS a 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = a.series_id
WHERE data_type_code = '10'
-- WHERE data_type_text = 'Women employees'

(1 row affected)

Total execution time: 00:00:00.113

WOMEN ACROSS INDUSTRIES 1125490

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;</span>

In [17]:
SELECT ROUND(SUM(a.VALUE), 0) AS 'PRODUCTION/NONSUPERVISORY EMPLOYEES'
FROM LaborStatisticsDB.dbo.annual_2016 AS a 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = a.series_id
WHERE data_type_code = '6'

--  data_type_text = 'PRODUCTION AND NONSUPERVISORY EMPLOYEES'

(1 row affected)

Total execution time: 00:00:00.113

PRODUCTION/NONSUPERVISORY EMPLOYEES 1263650

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?</span>

In [45]:
SELECT ROUND(AVG(j.value), 0) AS 'HOURS PRODUCTION/NONSUPERVISORY EMPLOYEES'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
WHERE  j.series_id LIKE '%07'


(1 row affected)

Total execution time: 00:00:00.178

HOURS PRODUCTION/NONSUPERVISORY EMPLOYEES 36

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please.</span>

In [1]:
SELECT ROUND(SUM(j.value), 0) AS 'PAYROLL PRODUCTION/NONSUPERVISORY EMPLOYEES'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
WHERE  j.series_id LIKE '%08'

(1 row affected)

Total execution time: 00:00:00.219

PAYROLL PRODUCTION/NONSUPERVISORY EMPLOYEES 48405

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [2]:
SELECT i.industry_name, ROUND(MAX(j.value), 0) AS 'HIGHESTHOURS'
FROM LaborStatisticsDB.dbo.january_2017 as j
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code
WHERE j.series_id LIKE '%07'
GROUP BY i.industry_name
ORDER BY HIGHESTHOURS DESC;

SELECT i.industry_name, ROUND(MIN(j.value), 0) AS 'LOWESTHOURS'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = j.series_id 
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code 
WHERE  j.series_id LIKE '%07'
GROUP BY industry_name
ORDER BY LOWESTHOURS ASC;

-- SELECT industry_name 
-- GROUP BY 


(501 rows affected)

(501 rows affected)

Total execution time: 00:00:00.724

industry_name HIGHESTHOURS Motor vehicle power train components 50 Motor vehicle metal stamping 47 NULL 47 Coal mining 47 Crushed and broken limestone mining 47 Oil and gas pipeline construction 47 Stone mining and quarrying 47 Support activities for mining 47 Support activities for oil and gas operations 47 Petroleum and coal products 46 Boilers 46 Mining 46 Mining and logging 46 Motor vehicle parts 45 Motor vehicles 45 Basic chemicals 45 Pipeline transportation 45 Power and communication system construction 45 Rubber products 45 Waste treatment and disposal 45 Transportation equipment 44 Turbine and power transmission equipment 44 Utility system construction 44 Surgical appliances and supplies 44 Special tools 44 Power generation and supply 44 Pulp 44 Oil and gas extraction 44 Other specialized trucking 44 Other heavy construction 44 Cement and concrete products 44 Aerospace products and parts 44 All other motor vehicle parts 44 Animal slaughtering and processing 44 Corrugated and solid fiber boxes 44 Electric power transmission and distribution 44 Motor vehicle gasoline engine and parts 44 Metal valves 44 Fossil fuel electric power generation 44 Frozen food 44 Heavy and civil engineering construction 44 Highway 44 Industrial machinery 44 HVAC and commercial refrigeration equipment 43 Forging and stamping 43 Fruit and vegetable preserving and specialty 43 General freight trucking 43 Metalworking machinery 43 Industrial building 43 Machinery 43 Electric power generation 43 Electrical equipment 43 Electrical equipment and appliances 43 Construction equipment 43 Aircraft 43 Other building equipment contractors 43 Other general purpose machinery 43 Other electrical equipment and components 43 Other fabricated metal products 43 Paper and paper products 43 Paperboard containers 43 Primary metals 43 Poultry processing 43 Plastics and rubber products 43 Petroleum 43 Sawmills and wood preservation 43 Turned products and screws 43 Utilities 43 Warehousing and storage 42 Water 42 Water and sewer system construction 42 Wood containers and pallets 42 Wood kitchen cabinets and countertops 42 Wood products 42 Truck transportation 42 Resin 42 Semiconductors and related devices 42 Sheet metal work 42 Plastics packaging materials 42 Plastics products 42 Plate work and fabricated structural products 42 Plywood and engineered wood products 42 Ready-mix concrete 42 Other wood products 42 Packaging and labeling services 42 Other plastics products 42 Ornamental and architectural metal products 42 Agricultural 42 All other wood products 42 All other fabricated metal products 42 Architectural and structural metals 42 Chemicals 42 Converted paper products 42 Durable goods 42 Electronic instruments 42 Fabricated metal products 42 Fabricated structural metal products 42 Lumber and wood 42 Machine shops 42 Machine shops and threaded products 42 Manufacturing 42 Logging 42 Millwork 42 Medical equipment and supplies 42 Natural gas distribution 42 Nonmetallic mineral products 42 General warehousing and storage 42 Fruits and vegetables 42 Fruit and vegetable canning and drying 42 Food manufacturing 42 Heavy machinery rental and leasing 42 Foundries 41 Goods-producing 41 Furniture and related products 41 Motor vehicle bodies and trailers 41 Lumber and construction supplies 41 Electrical equipment and wiring 41 Dairy products 41 Direct insurers 41 Computer and electronic products 41 Computer and software 41 Business to business electronic markets 41 Coating 41 Commercial and service industry machinery 41 Nondurable goods 41 Paper bags and coated and treated paper 41 Recyclable materials 41 Semiconductors and electronic components 41 Textile mills 41 Wood windows and doors 41 Wired telecommunications carriers 41 Waste management and remediation services 41 Specialized freight trucking 41 Tile and terrazzo contractors 41 Steel and precast concrete contractors 40 Telecommunications 40 Soaps 40 Waste collection 40 Upholstered household furniture 40 Site preparati

industry_name LOWESTHOURS Fitness and recreational sports centers 17 Family clothing stores 18 Civic and social organizations 18 Women's clothing stores 18 Skiing facilities 19 Clothing stores 19 NULL 19 Bowling centers 20 Clothing accessories stores 21 Clothing and clothing accessories stores 21 Drinking places 21 Amusements 22 Gift 22 Snack and nonalcoholic beverage bars 22 Shoe stores 23 Other personal care services 23 Arts 23 Book stores and news dealers 24 Food services and drinking places 24 Full-service restaurants 24 Limited-service restaurants 24 Leisure and hospitality 24 Other home furnishings stores 24 Restaurants and other eating places 24 Performing arts and spectator sports 25 Performing arts companies 25 Personal care services 25 Pet care services 25 Special food services 25 Sporting goods 25 Museums 25 Office supplies 25 Golf courses and country clubs 25 Hair 25 Hobby 25 Accommodation and food services 25 Florists 25 Coin-operated laundries and drycleaners 26 Beer 26 Cafeterias 26 Car washes 26 Sporting goods and musical instrument stores 26 Sporting goods stores 26 Offices of chiropractors 26 Offices of dentists 27 Offices of mental health practitioners 27 Personal and laundry services 27 Other personal services 27 RV parks and recreational camps 27 Services for the elderly and persons with disabilities 27 Social science and humanities research 27 Spectator sports 27 Home furnishings stores 27 Janitorial services 27 Food service contractors 27 Motion picture and sound recording industries 27 Miscellaneous store retailers 28 Marketing research and public opinion polling 28 Freestanding emergency medical centers 28 General merchandise stores 28 Independent artists 28 Individual and family services 28 Grocery stores 28 Food and beverage stores 28 Specialty food stores 28 Supermarkets and other grocery stores 28 Parking lots and garages 28 Offices of optometrists 28 Offices of other health practitioners 28 Offices of all other health practitioners 28 Offices of specialty therapists 28 Offices of mental health physicians 29 Pet and pet supplies stores 29 Pharmacies and drug stores 29 Other miscellaneous store retailers 29 Social assistance 29 Services to buildings and dwellings 29 School and employee bus transportation 29 Traveler accommodation 29 Used merchandise stores 29 Veterinary services 29 Convenience stores 29 Death care services 29 Cosmetic and beauty supply stores 29 Accommodation 29 Health and personal care stores 29 Home centers 29 Home health care services 29 Hotels and motels 29 Office supplies and stationery stores 29 Membership associations and organizations 30 Lessors of nonresidential buildings 30 Hardware stores 30 Jewelry 30 Gasoline stations 30 Gasoline stations with convenience stores 30 Carpet and upholstery cleaning services 30 Child and youth services 30 Child day care services 30 Drycleaning and laundry services 30 Vocational rehabilitation services 30 Used household and office goods moving 30 Retail trade 30 Other automotive repair and maintenance 30 Other health and personal care stores 30 Other individual and family services 30 Other professional and technical services 31 Other services 31 Photographic services 31 Radio broadcasting 31 Support activities for water transportation 31 Community care facilities for the elderly 31 Building material and garden supply stores 31 Building material and supplies dealers 31 Assisted living facilities for the elderly 31 Ambulatory health care services 31 Grantmaking foundations 31 Furniture and home furnishings stores 31 Labor unions and similar labor organizations 31 Human rights organizations 31 Health care and social assistance 32 Framing contractors 32 Landscaping services 32 Lawn and garden equipment and supplies stores 32 Marinas 32 Nursing and residential care facilities 32 Nursing care facilities 32 All other support services 32 Air transportation 32 Airport operations 32 Convention and trade show organizers 32 Continuing care retirement co

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [8]:

-- SELECT i.industry_name, ROUND(MAX(j.value), 0) AS 'HIGHESTPAYROLL'
-- FROM LaborStatisticsDB.dbo.january_2017 as j 
-- JOIN LaborStatisticsDB.dbo.series as s 
-- ON j.series_id = s.series_id
-- JOIN LaborStatisticsDB.dbo.industry as i 
-- ON i.industry_code = s.industry_code
-- WHERE  j.series_id LIKE '%82'
-- GROUP BY industry_name
-- ORDER BY HIGHESTPAYROLL

-- SELECT i.industry_name, ROUND(MIN(j.value), 0) AS 'LOWESTPAYROLL'
-- FROM LaborStatisticsDB.dbo.january_2017 as j 
-- JOIN LaborStatisticsDB.dbo.series as s 
-- ON j.series_id = s.series_id
-- JOIN LaborStatisticsDB.dbo.industry as i 
-- ON i.industry_code = s.industry_code
-- WHERE  j.series_id LIKE '%82'
-- GROUP BY industry_name
-- ORDER BY LOWESTPAYROLL



SELECT i.industry_name, ROUND(SUM(j.value), 0) AS 'HIGHESTPAYROLL'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code
WHERE j.series_id LIKE '%30'
GROUP BY industry_name
ORDER BY HIGHESTPAYROLL DESC;

SELECT i.industry_name, ROUND(SUM(j.value), 0) AS 'LOWESTPAYROLL'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code
WHERE j.series_id LIKE '%30'
GROUP BY industry_name
ORDER BY LOWESTPAYROLL

(501 rows affected)

(501 rows affected)

Total execution time: 00:00:00.326

industry_name HIGHESTPAYROLL NULL 119157 General freight trucking 17856 Mining 10143 Motor vehicles 8459 Industrial machinery 8204 Offices of physicians 8150 Chemicals 7984 Durable goods 7556 Other specialized trucking 7546 Reinsurance carriers 7242 Petroleum and coal products 7210 Fossil fuel electric power generation 7113 Paper and paper products 7104 Pipeline transportation 6934 Electric power generation 6907 Software publishers 6757 Power generation and supply 6654 Aircraft 6604 Custom computer programming services 6517 Physical 6517 Computer systems design services 6459 Electric power transmission and distribution 6435 Computer systems design and related services 6426 Nondurable goods 6383 Aerospace products and parts 6358 Portfolio management 6349 Utilities 6200 Research and development in the physical 6193 Oil and gas extraction 6106 Scientific research and development services 5920 Computer and software 5920 Coal mining 5819 Other financial investment activities 5758 Other heavy construction 5698 Oil and gas pipeline construction 5679 Securities brokerage 5646 Natural gas distribution 5529 Administrative management consulting services 5439 Basic chemicals 5410 Biotechnology research 5325 Semiconductors and related devices 5320 Investment advice 5171 Power and communication system construction 5155 Architectural and engineering services 5120 Offices of lawyers 5064 Wholesale trade agents and brokers 5055 Advertising agencies 5049 Mining and logging 5043 Publishing industries 5024 Data processing 5014 Professional and technical services 5011 Management consulting services 5003 Electronic markets and agents and brokers 5002 Management and technical consulting services 5001 Legal services 4936 Support activities for mining 4924 Insurance carriers 4876 Direct insurers 4876 Architectural services 4870 Utility system construction 4863 Real estate credit 4856 Pulp 4849 Commercial equipment 4845 Motor vehicle power train components 4822 Waste treatment and disposal 4817 Direct life and health insurance carriers 4811 Other building equipment contractors 4806 Heavy and civil engineering construction 4789 Industrial building 4786 Electronic instruments 4768 Steel and precast concrete contractors 4739 Heavy machinery rental and leasing 4738 Support activities for oil and gas operations 4736 Public relations agencies 4723 Sporting goods 4661 Other information services 4609 Telecommunications 4589 Wired telecommunications carriers 4549 Electrical contractors 4524 Land subdivision 4497 Insurance carriers and related activities 4473 Nonresidential building 4465 Medical equipment 4459 General medical and surgical hospitals 4452 Financial transaction processing and clearing 4419 Information 4397 Hospitals 4379 Transportation equipment 4377 Commercial building 4368 Resin 4355 Commercial and service industry machinery 4354 Highway 4349 Office equipment 4343 Construction equipment 4340 Periodical publishers 4331 Grantmaking foundations 4325 Other insurance-related activities 4324 Tile and terrazzo contractors 4315 Building equipment contractors 4310 Marketing consulting services 4303 Management of companies and enterprises 4293 Graphic design services 4290 Advertising and related services 4272 Air transportation 4268 Managing offices 4266 Professional organizations 4265 Electric goods 4248 Support activities for water transportation 4246 Turbine and power transmission equipment 4207 Labor unions and similar labor organizations 4179 Electrical equipment and wiring 4167 Process and logistics consulting services 4157 Third-party administration of insurance funds 4145 Drywall and insulation contractors 4115 Nondepository credit intermediation 4108 Motor vehicle gasoline engine and parts 4101 Water 4099 Special tools 4094 Water and sewer system construction 4084 Business associations 4070 Computer and electronic products 4066 Specialized design services 4054 Construction 4053 Plumbing and HVAC contractors 4049 Petroleum 4036 Machinery and equipm

industry_name LOWESTPAYROLL Bowling centers 939 Family clothing stores 976 Fitness and recreational sports centers 979 Women's clothing stores 983 Limited-service restaurants 1009 Clothing stores 1033 Snack and nonalcoholic beverage bars 1044 Cafeterias 1089 Civic and social organizations 1093 Gift 1102 Clothing accessories stores 1130 Drinking places 1167 Restaurants and other eating places 1167 Food services and drinking places 1187 Clothing and clothing accessories stores 1189 Skiing facilities 1199 Florists 1212 Book stores and news dealers 1250 Convenience stores 1253 Accommodation and food services 1257 Car washes 1261 Amusements 1275 Leisure and hospitality 1294 Full-service restaurants 1312 Beer 1320 Coin-operated laundries and drycleaners 1324 Gasoline stations with convenience stores 1347 RV parks and recreational camps 1357 Specialty food stores 1360 Janitorial services 1366 Other home furnishings stores 1373 Gasoline stations 1383 Pet care services 1398 Food and beverage stores 1440 Shoe stores 1441 Special food services 1443 Used merchandise stores 1446 Grocery stores 1453 Hobby 1457 Supermarkets and other grocery stores 1461 General merchandise stores 1462 Services for the elderly and persons with disabilities 1474 Office supplies 1485 Golf courses and country clubs 1513 Miscellaneous store retailers 1521 Parking lots and garages 1533 Pet and pet supplies stores 1539 Sporting goods and musical instrument stores 1539 Other personal services 1541 Sporting goods stores 1546 Other automotive repair and maintenance 1550 Food service contractors 1551 Arts 1562 Child day care services 1567 Cosmetic and beauty supply stores 1597 Hardware stores 1624 Individual and family services 1652 Other miscellaneous store retailers 1658 Social assistance 1659 Other personal care services 1660 Other gasoline stations 1663 Assisted living facilities for the elderly 1663 Personal care services 1665 Hair 1670 Accommodation 1687 Hotels and motels 1688 Vocational rehabilitation services 1689 Traveler accommodation 1698 Personal and laundry services 1701 Home centers 1726 Airport operations 1775 Services to buildings and dwellings 1796 Home furnishings stores 1806 Retail trade 1809 Residential intellectual and developmental disability facilities 1839 Community care facilities for the elderly 1840 Other health and personal care stores 1892 Offices of chiropractors 1892 Other ground passenger transportation 1903 Carpet and upholstery cleaning services 1920 Seafood product preparation and packaging 1949 Jewelry 1953 Building material and supplies dealers 1970 Building material and garden supply stores 1977 Apparel 1979 Telephone call centers 1996 Continuing care retirement communities 1997 Residential mental health facilities 2014 Used household and office goods moving 2048 Office supplies and stationery stores 2049 Nursing and residential care facilities 2053 School and employee bus transportation 2066 Health and personal care stores 2070 Security and armored car services 2076 Lawn and garden equipment and supplies stores 2079 Home health care services 2087 Other residential care facilities 2089 Linen and uniform supply 2101 Offices of optometrists 2111 Offices of mental health practitioners 2136 Nursery stock and florists' supplies 2141 Optical goods stores 2150 Textile product mills 2180 Nursing care facilities 2189 Pharmacies and drug stores 2190 Furniture and home furnishings stores 2192 Gambling industries 2193 Other individual and family services 2197 Child and youth services 2199 All other support services 2206 Emergency and other relief services 2234 Household appliance stores 2240 Poultry processing 2241 Temporary help services 2244 Investigation and security services 2246 Other services to buildings and dwellings 2254 Veterinary services 2271 Household goods repair and maintenance 2271 Landscaping services 2290 Transit and ground passenger transportation 2291 Business support services 2293 Death care services 2319 Administrative an

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [2]:
SELECT j.series_id, j.value, s.industry_code, i.industry_name
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id 
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code 
WHERE j.series_id LIKE '%82' AND j.value > (
    SELECT AVG(a.value) 
    FROM LaborStatisticsDB.dbo.annual_2016 as a 
    WHERE a.series_id LIKE '%82'
)

--For subqueries #1, when you get to the where statement, try using data type code 82 AND the value of january.2017 and then have your greater than statement for the average of annual_2016.value.

(294 rows affected)

Total execution time: 00:00:00.269

series_id value industry_code industry_name CES0500000082 74498171 05000000 Total private CES0500000082 74498171 05000000 Total private CEU0500000082 73474302 05000000 Total private CEU0500000082 73474302 05000000 Total private CES0600000082 13426820 06000000 Goods-producing CES0600000082 13426820 06000000 Goods-producing CEU0600000082 12711945 06000000 Goods-producing CEU0600000082 12711945 06000000 Goods-producing CES0800000082 60914722 08000000 Private service-providing CES0800000082 60914722 08000000 Private service-providing CEU0800000082 60807771 08000000 Private service-providing CEU0800000082 60807771 08000000 Private service-providing CES2000000082 5335274 20000000 Construction CES2000000082 5335274 20000000 Construction CEU2000000082 4724156 20000000 Construction CEU2000000082 4724156 20000000 Construction CES2023600082 1008035 20236000 Construction of buildings CES2023600082 1008035 20236000 Construction of buildings CEU2023600082 940800 20236000 Construction of buildings CEU2023600082 940800 20236000 Construction of buildings CES2023700082 908374 20237000 Heavy and civil engineering construction CES2023700082 908374 20237000 Heavy and civil engineering construction CES2023800082 3421902 20238000 Specialty trade contractors CES2023800082 3421902 20238000 Specialty trade contractors CEU2023800082 3062211 20238000 Specialty trade contractors CEU2023800082 3062211 20238000 Specialty trade contractors CES2023820082 1696080 20238200 Building equipment contractors CES2023820082 1696080 20238200 Building equipment contractors CEU2023820082 1621266 20238200 Building equipment contractors CEU2023820082 1621266 20238200 Building equipment contractors CES3000000082 7491723 30000000 Manufacturing CES3000000082 7491723 30000000 Manufacturing CEU3000000082 7411960 30000000 Manufacturing CEU3000000082 7411960 30000000 Manufacturing CES3100000082 4859716 31000000 Durable goods CES3100000082 4859716 31000000 Durable goods CEU3100000082 4808368 31000000 Durable goods CEU3100000082 4808368 31000000 Durable goods CES3133200082 871298 31332000 Fabricated metal products CES3133200082 871298 31332000 Fabricated metal products CEU3133200082 867242 31332000 Fabricated metal products CEU3133200082 867242 31332000 Fabricated metal products CES3133600082 1257764 31336000 Transportation equipment CES3133600082 1257764 31336000 Transportation equipment CEU3133600082 1242685 31336000 Transportation equipment CEU3133600082 1242685 31336000 Transportation equipment CES3200000082 2633628 32000000 Nondurable goods CES3200000082 2633628 32000000 Nondurable goods CEU3200000082 2605069 32000000 Nondurable goods CEU3200000082 2605069 32000000 Nondurable goods CES3231100082 851406 32311000 Food manufacturing CES3231100082 851406 32311000 Food manufacturing CEU3231100082 842201 32311000 Food manufacturing CEU3231100082 842201 32311000 Food manufacturing CES4000000082 14850661 40000000 Trade CES4000000082 14850661 40000000 Trade CEU4000000082 14732108 40000000 Trade CEU4000000082 14732108 40000000 Trade CES4142000082 4480134 41420000 Wholesale trade CES4142000082 4480134 41420000 Wholesale trade CEU4142000082 4506110 41420000 Wholesale trade CEU4142000082 4506110 41420000 Wholesale trade CES4142300082 2261006 41423000 Durable goods CES4142300082 2261006 41423000 Durable goods CEU4142300082 2283094 41423000 Durable goods CEU4142300082 2283094 41423000 Durable goods CES4142400082 1367413 41424000 Nondurable goods CES4142400082 1367413 41424000 Nondurable goods CEU4142400082 1348136 41424000 Nondurable goods CEU4142400082 1348136 41424000 Nondurable goods CES4142500082 855291 41425000 Electronic markets and agents and brokers CES4142500082 855291 41425000 Electronic markets and agents and brokers CEU4142500082 873578 41425000 Electronic markets and agents and brokers CEU4142500082 873578 41425000 Electronic markets and agents and brokers CES4142512082 831319 41425120 Wholesale trade agents and brokers CES4142512082 831319 41425120 Wholesale trade agents and br

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [6]:
-- SELECT j.series_id, j.value, j.year, j.period 
-- FROM LaborStatisticsDB.dbo.january_2017 as j 
-- WHERE j.series_id LIKE '%30'
-- UNION
-- SELECT a.series_id, a.value, a.year, a.period  
-- FROM LaborStatisticsDB.dbo.annual_2016 as a 
-- WHERE a.series_id LIKE '%30'



-- ON j.series_id = s.series_id 
-- JOIN LaborStatisticsDB.dbo.industry as i 
-- ON i.industry_code = s.industry_code 
-- WHERE j.value > (
--     SELECT AVG(a.value) 
--     FROM LaborStatisticsDB.dbo.annual_2016 as a 
--     WHERE a.series_id LIKE '%82'

-- data_type_code = 6



SELECT ROUND(AVG(annual_2016.value), 2), MAX(annual_2016.year) AS Year
FROM LaborStatisticsDB.dbo.annual_2016
Where annual_2016.series_id LIKE '%30'
GROUP BY annual_2016.period
UNION
SELECT ROUND(AVG(january_2017.value), 2), MAX(january_2017.year) AS Earnings
FROM LaborStatisticsDB.dbo.january_2017
Where january_2017.series_id LIKE '%30'



(2 rows affected)

Total execution time: 00:00:00.269

(No column name) Year 797.2 2016 808.53 2017

In [53]:
SELECT AVG(j.value) AS 'JANUARY 2017'
FROM LaborStatisticsDB.dbo.january_2017 as j 
WHERE j.series_id LIKE '%30'
-- JOIN 

SELECT AVG(a.value) AS 'ANNUAL 2016'
FROM LaborStatisticsDB.dbo.annual_2016 as a 
-- ON a.series_id, a.value, a.year, a.period 
WHERE a.series_id LIKE '%30'

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.188

JANUARY 2017 808.5303811252254

ANNUAL 2016 797.1962613430123

In [72]:
SELECT SUM(j.value) as 'JANUARY FARE'
FROM LaborStatisticsDB.dbo.january_2017 as j 
WHERE j.period = 'M01' AND j.series_id LIKE '%30'


SELECT SUM(a.value) as 'ANNUAL FARE'
FROM LaborStatisticsDB.dbo.annual_2016 as a
WHERE a.period = 'M13' AND a.series_id LIKE '%30'

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.224

JANUARY FARE 1782000.9599999967

ANNUAL FARE 878510.2799999996

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

During Period 'M01' or January, is when the production and nonsupervisory employees fare better as seen with the following code: 

  

<span style="color: #0000ff;">SELECT</span> <span style="color: #795e26;">SUM</span>(j.value) <span style="color: #0000ff;">as</span> <span style="color: #a31515;">'JANUARY FARE'</span>

<span style="color: #0000ff;">FROM</span> LaborStatisticsDB.dbo.january\_2017 <span style="color: #0000ff;">as</span> j

<span style="color: #0000ff;">WHERE</span> j.period <span style="color: #000000;">=</span> <span style="color: #a31515;">'M01'</span> <span style="color: #0000ff;">AND</span> j.series\_id <span style="color: #0000ff;">LIKE</span> <span style="color: #a31515;">'%30'</span>

  
  

<span style="color: #0000ff;">SELECT</span> <span style="color: #795e26;">SUM</span>(a.value) <span style="color: #0000ff;">as</span> <span style="color: #a31515;">'ANNUAL FARE'</span>

<span style="color: #0000ff;">FROM</span> LaborStatisticsDB.dbo.annual\_2016 <span style="color: #0000ff;">as</span> a

<span style="color: #0000ff;">WHERE</span> a.period <span style="color: #000000;">=</span> <span style="color: #a31515;">'M13'</span> <span style="color: #0000ff;">AND</span> a.series\_id <span style="color: #0000ff;">LIKE</span> <span style="color: #a31515;">'%30'</span><span style="color: rgb(0, 0, 255);">SELECT</span> <span style="color: rgb(121, 94, 38);">SUM</span>(j.value) <span style="color: rgb(0, 0, 255);">as</span> <span style="color: rgb(163, 21, 21);">'JANUARY FARE'</span>

<span style="color: #0000ff;">FROM</span> LaborStatisticsDB.dbo.january\_2017 <span style="color: #0000ff;">as</span> j

<span style="color: #0000ff;">WHERE</span> j.period <span style="color: #000000;">=</span> <span style="color: #a31515;">'M01'</span> <span style="color: #0000ff;">AND</span> j.series\_id <span style="color: #0000ff;">LIKE</span> <span style="color: #a31515;">'%30'</span>

  
  

which shows that January's fare was 1782000.96 and the 'M13' period totaled 878510.28

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better? The industries with the highest pay include "Total Private", "Private Service-Providing" and "Professional and business services."</span>

Double click here to add your answer.

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

<span style="font-size: 13.3333px; white-space: pre-wrap;">It would have been nice to have the foreign keys and primary keys listed out for us as well as a table/spreadsheet with the January_2017 and Annual_2016 infomation.</span>